.github/workflows/ml-cicd.yml

```
@@ -180,73 +149,36 @@
-        with:
-          name: trained-models
-          path: |
-            src/ml/models/trained/house_price_prediction.pkl
-            src/ml/models/trained/preprocessor.pkl
-          retention-days: 30
+          pytest src/tests/ -v --cov=src --cov-report=xml --cov-report=html
 
-      - name: Upload processed data
-        uses: actions/upload-artifact@v4
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)